torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -1221,14 +1220,48 @@ void LLVMCodeGenImpl::visit(const BaseCallNode* v) {
 
 static void applyMathFunctionAttributes(llvm::Function* f) {
   f->addFnAttr(llvm::Attribute::ReadNone);
-  f->addFnAttr(llvm::Attribute::NoFree);
   f->addFnAttr(llvm::Attribute::NoUnwind);
   // TODO: Adding this attr should be correct, but as of LLVM 9.0.1 adding it
   // causes some math functions to incorrectly be turned into tail calls.
   // f->addFnAttr(llvm::Attribute::Speculatable);
+#if LLVM_VERSION_MAJOR == 9
+  f->addFnAttr(llvm::Attribute::NoFree);
   f->addFnAttr(llvm::Attribute::WillReturn);
+#endif
 }
 
+namespace {
+#if LLVM_VERSION_MAJOR == 9
```

Comment:
Could you please add to PR description why we are supporting both versions 8 and 9?

Reply:
Is the PR description not clear on that?  I guess it mentions llvm 7, not 8 (FB's internal build defines LLVM_VERSION_MAJOR as 8, but the OrcJIT API is clearly 7-ish...)

Reply:
that makes sense! sorry my bad